File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema14Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*

class Schema14Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema08Test::class.java.getResourceAsStream("/schema14.dbs"))
    }

    @Test
    fun test() {
        val message = mapOf(
                "uuid" to UUID.randomUUID().toString(),
                "payload" to mapOf(
                        ":type" to 0,
                        "identityId" to UUID.randomUUID().toString(),
                        "eventName" to "test",
                        "eventData" to mapOf("a" to "b")
                ),
                "replyTo" to "asdf",
                "timeout" to 1000
                )
        val result = engine.deserialize(engine.serialize(message))

        assertMap(message, result)
    }

}

File: dynabuffers-python/tests/usecase/Schema14Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema13Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema14.dbs"))

    def test(self):
        obj = {"uuid": "1234",
               "payload": {":type": 0, "identityId": "1234", "eventName": "test", "eventData": {"a": "b"}},
               "replyTo": "test",
               "timeout": 1000}
        result = self.engine.deserialize(self.engine.serialize(obj))
        self.assertEqual(result, obj)


if __name__ == "__main__":
    unittest.main()
